django/db/models/fields/related.py

```diff
@@ -1375,6 +1376,22 @@ def _check_relationship_model(self, from_model=None, **kwargs):
 
         return errors
 
+    def _check_table_uniqueness(self, **kwargs):
+        registered_tables = {
+            model._meta.db_table: model
+            for model in self.opts.apps.get_models(include_auto_created=True)
```

Comment:
Shouldn't you exclude `self.rel.through` from here? Else your automatically created table will always be present in `registered_tables`.